capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -132,7 +132,9 @@ public static JsonObject createRequestBodyJson(HttpServletRequest request) throw
       return null;
     }
     String parameterValue = entry.getValue()[0];
-    Field[] fields = genericClass.getMyType().getDeclaredFields();
+    Class superClass = genericClass.getMyType().getSuperclass();
+    List<Field> fields = new ArrayList<>(Arrays.asList(genericClass.getMyType().getDeclaredFields()));
+    fields.addAll(Arrays.asList(superClass.getDeclaredFields()));
```

Comment:
I'm glad this get superclass declared fields works!

Reply:
Me too! Lets us deal with our Group hierarchy nicely.

Reply:
If we have "deeper" hierarchies, this may fail. Doesn't seem to be an issue for now but worth looking into in the future.